org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/RecordsRestrictedClassTest.java

```diff
@@ -3746,22 +3746,18 @@ public void testBug562439_019() throws IOException, ClassFormatException {
 			"    RuntimeInvisibleAnnotations: \n" +
 			"      #8 @Annot(\n" +
 			"      )\n" +
-			"    RuntimeInvisibleTypeAnnotations: \n" +
-			"      #8 @Annot(\n" +
-			"        target type = 0x13 FIELD\n" +
-			"      )\n" +
 			"  \n";
```

Comment:
How do changes in this test suite relate to #1092? You wrote that one is orthogonal, yet here you are fixing duplication. Just curious.

Reply:
> How do changes in this test suite relate to #1092? You wrote that one is orthogonal, yet here you are fixing duplication. Just curious.

This is reversing the effect of https://bugs.eclipse.org/bugs/show_bug.cgi?id=552082 - In the absence of explicit target @Annot is both a type annotation and a SE7 declaration annotation and gets emitted once each as a declarative annotation and as a type annotation.

In https://bugs.eclipse.org/bugs/show_bug.cgi?id=552082#c1, you asserted 
```
This change made me look also at this in existing JLS §9.7.4:
   "If the annotation's type is applicable in the declaration context corresponding to the declaration and in type contexts, then the annotation is deemed to apply to both the declaration and the type which is closest to the annotation."

I don't see this dual applicability represented in the class files generated by javac nor ecj. Just mentioning as am NOT going to fix this, i.e., I will not invest efforts to become incompatible with javac in this case.
```

That is not so! We do implement the dual applicability and that needs to adapt to the latest change since in the absence of a target @Annot is not a type annotation *anymore*

The duplication we are talking about in https://github.com/eclipse-jdt/eclipse.jdt.core/issues/1092 is indeed orthogonal brought about by (incorrect) double traversal of the relevant AST nodes.